react-file-uploads/server/server.ts

```diff
@@ -0,0 +1,79 @@
+import 'dotenv/config';
+import express from 'express';
+import pg from 'pg';
+import {
+  ClientError,
+  errorMiddleware,
+  uploadsMiddleware,
+} from './lib/index.js';
+
+type Image = {
+  imageId: number;
+  url: string;
+  caption: string;
+};
+
+const db = new pg.Pool({
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+
+app.use(express.static('public'));
+app.use(express.json());
+
+app.post(
+  '/api/uploads',
+  uploadsMiddleware.single('image'),
+  async (req, res, next) => {
+    try {
+      if (!req.file) throw new ClientError(400, 'no file field in request');
+      const { caption } = req.body as Partial<Image>;
+      if (!caption) {
+        throw new ClientError(400, 'caption is a required field');
+      }
+      const imageUrl = '/images/' + req.file.filename;
+      if (!imageUrl) throw new ClientError(400, 'url is a required field');
```

Comment:
You just assigned a value to `imageUrl` on the previous line, so it will never be undefined.